packages/blockly-field-slider/src/field_slider.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Number slider input field.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+import * as Blockly from 'blockly/core';
+
+export class FieldSlider extends Blockly.FieldNumber {
+  /**
+   * Class for an number slider field.
+   * @param {string|number=} opt_value The initial value of the field. Should
+   *    cast to a number. Defaults to 0.
+   * @param {?(string|number)=} opt_min Minimum value.
+   * @param {?(string|number)=} opt_max Maximum value.
+   * @param {?(string|number)=} opt_precision Precision for value.
+   * @param {?Function=} opt_validator A function that is called to validate
+   *    changes to the field's value. Takes in a number & returns a validated
+   *    number, or null to abort the change.
+   * @param {Object=} opt_config A map of options used to configure the field.
+   *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/number#creation}
+   *    for a list of properties this parameter supports.
+   * @extends {Blockly.FieldNumber}
+   * @constructor
+   */
+  constructor(opt_value, opt_min, opt_max, opt_precision,
+      opt_validator, opt_config) {
+    super(opt_value, opt_min, opt_max, opt_precision, opt_validator, opt_config);
+
+    /**
+     * Array holding info needed to unbind events.
+     * Used for disposing.
+     * Ex: [[node, name, func], [node, name, func]].
+     * @type {!Array.<Array<?>>}
+     * @private
+     */
+    this.boundEvents_ = [];
+
+    /**
+     * The HTML range input element.
+     * @type {?HTMLInputElement}
+     * @private
+     */
+    this.sliderInput_ = null;
+  }
+
+  /**
+   * Constructs a FieldSlider from a JSON arg object.
+   * @param {!Object} options A JSON object with options (value, min, max, and
+   *                          precision).
+   * @return {!FieldSlider} The new field instance.
+   * @package
+   * @nocollapse
+   */
+  static fromJson (options) {
+    return new FieldSlider(options['value'],
+        undefined, undefined, undefined, undefined, options);
+  }
+
+  /**
+   * Show the inline free-text editor on top of the text along with the slider
+   *    editor.
+   * @param {Event=} opt_e Optional mouse event that triggered the field to open,
+   *     or undefined if triggered programmatically.
+   * @param {boolean=} opt_quietInput True if editor should be created without
```

Comment:
No longer need this paramter.

Reply:
Updated param with "_" to indicate as unused (so that there aren't issues with missing param in overriden method).